### PR TITLE
Allow customizing module-layer Hiera configuration

### PR DESCRIPTION
**Pull request by sanfrancrisko**
This adds documentation to the work in  #732 .

Closes #732.

### CODE REVIEW

File: docs/documentation/configuration/index.md [hunk 1]
```diff
@@ -227,3 +227,35 @@ unnecessary in a testing environment.
 
 The search path for global modules. Almost always unnecessary in a testing
 environment.
+
+### disable_module_hiera
+**Type:** Boolean<br /> 
+**Default:** `false`<br />
+**Puppet Version(s):** 6.x
```
Comment (nabertrand) [inline docs/documentation/configuration/index.md:234]: I believe these features should work with Puppet version >= 4.9.0, not just 6.x

File: lib/rspec-puppet/support.rb [hunk 2]
```diff
@@ -87,21 +87,18 @@ def load_catalogue(type, exported = false, manifest_opts = {})
...
-          node_params: node_params_hash,
-          trusted_external_data: trusted_external_data_hash,
-          ignored_cache_params: {
-            hiera_data_value: hiera_data_value,
-            trusted_server_facts: RSpec.configuration.trusted_server_facts,
-          },
-        )
+(??)        catalogue = build_catalog(node_name, build_facts, trusted_facts_hash(node_name), hiera_config_value,
+(??)                                  build_code(type, manifest_opts), exported, node_params_hash, hiera_data_value,
+(??)                                  RSpec.configuration.trusted_server_facts)
```
Comment (nabertrand) [inline lib/rspec-puppet/support.rb:101]: Did this section still need some work? It doesn't look like `rspec_config_values` gets used anywhere anymore.

**Review by nabertrand — COMMENTED** [review]
@sanfrancrisko thanks for adding the documentation.

**Comment by scotje** [comment]
@sanfrancrisko @DavidS I think this is the last thing blocking the 2.8.0 release, is there anything I can do to help get this PR wrapped up?

**Comment by nabertrand** [comment]
I've resolved my above comments in my original branch and am working on adding additional long-form Hiera testing documentation as requested in the original PR. I think I'll just re-open #732 once I've finished the long-form documentation if that works.

File: lib/rspec-puppet/support.rb (hunk 2 above)
Comment (DavidS) [reply lib/rspec-puppet/support.rb]: fixed the underlying rebase

File: docs/documentation/configuration/index.md (hunk 1 above)
Comment (DavidS) [reply docs/documentation/configuration/index.md]: adjusted for all new config entries

**Comment by DavidS** [comment]
Rebased and conflict resolved; integrated fixes on which puppet versions are called out as compatible.

@nabertrand if you come around to finish the long-form docs work, please do so in a fresh PR so you don't have to deal with the conflicts since this PR already contains your code and conflict resolution.

**Pull request merged**